packages/core/crates/core-strategy/src/manager.rs

```diff
@@ -0,0 +1,9 @@
+use crate::traits::Strategy;
+
+pub struct StrategyManager {
+    strategies: Vec<Box<dyn Strategy>>
```

Comment:
Or you could create a derive macro to derive the aggregated strategy from the member strategies (remove the allocations and dereference) - e.g. what libp2p does with the `NetworkBehavior`.